File: leetcode/001_two_sum/sol.py
def twoSum(nums, target):
    """
    :type nums: List[int]
    :type target: int
    :rtype: List[int]
    """

    # Brute force
    # for i in range(0, len(nums)):
    #     for j in range(i+1, len(nums)):
    #         if nums[i] + nums[j] == target:
    #             return [i, j]

    # Hash Table
    my_dict = {}
    for i in range(0, len(nums)):
        if (target - nums[i]) in my_dict:
            if my_dict[target - nums[i]] != i or target - nums[i] != nums[i]:
                return [my_dict[target - nums[i]], i]
        my_dict[nums[i]] = i

print twoSum([3,2,4], 6)
print twoSum([3,3], 6)

File: leetcode/496_next_greater_element_i/sol.rb
def next_greater_element(find_nums, nums)
  my_dict = Hash.new
  nums.each_with_index do |el, i|
    if i == nums.length - 1
      my_dict[el] = -1
      break
    end

    j = i + 1

    while j <= nums.length - 1
      if nums[j] > el
        my_dict[el] = nums[j]
        break
      end

      my_dict[el] = -1 if j == nums.length - 1
      j += 1
    end
  end

  find_nums.map { |el| my_dict[el] }
end
